src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,134 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.RegistrationException;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private static StorageDaoImpl storageDao;
+    private static final String DEFAULT_LOGIN = "oleksii";
+    private static final String DEFAULT_PASSWORD = "12121212";
+    private static User user = new User();
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user.setLogin(DEFAULT_LOGIN);
+        user.setPassword(DEFAULT_PASSWORD);
+        user.setAge(27);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+    /*
+    Testing the login field
+     */
+
+    @Test
+    void register_loginDetailsAreCorrect_Ok() {
+        user.setLogin(DEFAULT_LOGIN);
+        User actual = registrationService.register(user);
+        assertEquals(user, actual);
+        User expected = storageDao.get(DEFAULT_LOGIN);
+        assertEquals(expected, actual);
+
+    }
+
+    @Test
+    void register_user_notOk() {
+        Storage.people.add(user);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void register_withNull_notOk() {
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(null));
+    }
+
+    @Test
+    void register_emptyLineLogin_notOk() {
+        user.setLogin("  ");
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void register_checkForAbsenceInTheFirstLetterOfCharacter_notOk() {
+        user.setLogin("#oleksii");
```

Comment:
checkForAbsenceInTheFirstLetterOfCharacter - I can not understand this
What case are you testing and why there is exception?

Reply:
here I wanted to check that the first element is a letter, and by exception they didn’t come up with anything better in all tests

Reply:
then name it like register_firstLoginCharacterIsNotLetter_notOk